doc/dox_comments/header_files/asn.h

```diff
@@ -0,0 +1,230 @@
+    \param certSz size of the certificate in bytes
+    \param heap pointer to heap hint for memory allocation (can be NULL)
+    \param cm pointer to certificate manager containing trusted CAs
+
+    _Example_
+    \code
+    byte cert[2048] = { /* DER encoded certificate */ };
+    word32 certSz = sizeof(cert);
+    WOLFSSL_CERT_MANAGER* cm;
+    
+    cm = wolfSSL_CertManagerNew();
+    wolfSSL_CertManagerLoadCA(cm, "ca-cert.pem", NULL);
+    
+    int ret = wc_CheckCertSignature(cert, certSz, NULL, cm);
+    if (ret == 0) {
+        // certificate signature is valid
+    }
+    wolfSSL_CertManagerFree(cm);
+    \endcode
+
+    \sa wolfSSL_CertManagerNew
+    \sa wolfSSL_CertManagerLoadCA
+*/
+int wc_CheckCertSignature(const byte* cert, word32 certSz, void* heap,
+                          void* cm);
+
+/*!
+    \ingroup ASN
+    \brief This function encodes an array of word16 values into an ASN.1
+    Object Identifier (OID) in DER format. OIDs are used to identify
+    algorithms, extensions, and other objects in certificates and
+    cryptographic protocols.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If in, inSz, or outSz are invalid.
+    \return BUFFER_E If out is not NULL and outSz is too small.
+
+    \param in pointer to array of word16 values representing OID components
+    \param inSz number of components in the OID
+    \param out pointer to buffer to store encoded OID (can be NULL to
+    calculate size)
+    \param outSz pointer to size of out buffer; updated with actual size
+
+    _Example_
+    \code
+    word16 oid[] = {1, 2, 840, 113549, 1, 1, 11}; // sha256WithRSAEncryption
+    byte encoded[32];
+    word32 encodedSz = sizeof(encoded);
+    
+    int ret = wc_EncodeObjectId(oid, sizeof(oid)/sizeof(word16),
+                                encoded, &encodedSz);
+    if (ret == 0) {
+        // encoded contains DER encoded OID
+    }
+    \endcode
+
+    \sa wc_BerToDer
+*/
+int wc_EncodeObjectId(const word16* in, word32 inSz, byte* out,
+                      word32* outSz);
+
+/*!
+    \ingroup ASN
+    \brief This function sets the algorithm identifier in DER format. It
+    encodes the algorithm OID and optional parameters based on the
+    algorithm type and curve size.
+
+    \return Length of the encoded algorithm identifier on success.
+    \return Negative value on error.
+
+    \param algoOID algorithm object identifier constant
+    \param output pointer to buffer to store encoded algorithm ID
+    \param type type of encoding (oidSigType, oidHashType, etc.)
+    \param curveSz size of the curve for ECC algorithms (0 for non-ECC)
+
+    _Example_
+    \code
+    byte algId[32];
+    word32 len;
+    
+    len = SetAlgoID(CTC_SHA256wRSA, algId, oidSigType, 0);
+    if (len > 0) {
+        // algId contains encoded algorithm identifier
+    }
+    \endcode
+
+    \sa wc_EncodeObjectId
+*/
+word32 SetAlgoID(int algoOID, byte* output, int type, int curveSz);
```

Comment:
Same... Another WOLFSSL_ASN_API. Please add note about it not normally being public unless WOLFSSL_ASN_API is used. Some customers may want direct access to internal ASN API's.